### PR TITLE
GitHub action

### PR DESCRIPTION
**Pull request by Bowenislandsong**
This commit enables Github Action. The uses of external images need sha based tag whereas GitHub action repos can use normal version tag.

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+on:
+  push:
+    branches: 
+      - master
+  pull_request:
+    branches: 
+      - master
+
+jobs:
+
```
Comment (awgreene) [inline .github/workflows/go.yml]: Please remove this empty line.

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: Go
```
Comment (awgreene) [inline .github/workflows/go.yml]: Please update this file type to `.yaml`

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+on:
+  push:
+    branches: 
+      - master
+  pull_request:
+    branches: 
+      - master
+
```
Comment (awgreene) [inline .github/workflows/go.yml]: Please remove this empty newline.

Comment (awgreene) [reply .github/workflows/go.yml]: Please do this in all places.

**Comment by benluddy** [comment]
/lgtm

I agree with @awgreene that the extra newlines aren't helping readability for me. Are there any sets of published YAML style rules with significant adoption?

File: .github/workflows/go.yml (hunk 1 above)
Comment (Bowenislandsong) [reply .github/workflows/go.yml]: lol, yeah, github propagated this format by default, so I went with it... lol. changing it,

File: .github/workflows/go.yml (hunk 2 above)
Comment (Bowenislandsong) [reply .github/workflows/go.yml]: this file type is .yaml. This is the name of the ... action .... won't it be weird to have an action called go.yaml?

Comment (Bowenislandsong) [reply .github/workflows/go.yml]: wait ... this type is .... yml ......

**Comment by Bowenislandsong** [comment]
> /lgtm
> 
> I agree with @awgreene that the extra newlines aren't helping readability for me. Are there any sets of published YAML style rules with significant adoption?

I mostly try to keep the default style that github provides. The main thing was to keep each test separate, but since you all don't like it,,, I changed it.

**Pull request merged**